### PR TITLE
Pre commit fixes

### PR DESCRIPTION
**Pull request by petemounce**

### CODE REVIEW

File: tasks/install-on-Windows.yml [hunk 1]
```diff
@@ -1,111 +1,116 @@
...
+      user_cannot_change_password: true
+    register: buildkite_agent_user
 
-# when a user is created, their profile directory only starts to exist after first process / logon as that user.
-# we definitely need this to exist for things like .ssh/* and friends.
-- name: make the user profile
-  win_user_profile:
-    username: "{{ buildkite_agent_user.sid }}"
-    state: present
+  - name: add account to Log on as a service
```
Comment (Helcaraxan) [inline tasks/install-on-Windows.yml:17]:
```suggestion
  - name: add account to log on as a service
```

**Review by Helcaraxan — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,17 @@
 ---
+after_script:
```
Comment (DoomGerbil) [inline .travis.yml:2]: Alpha-sort makes sense, but in this case, logically speaking, so would order-by-chronological-execution (eg `before_script`->`script`->`after_script`).

Something to think about perhaps, but I think I'm mostly fine with either order in this type of case.

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
+  - workflow/on-failure.sh
 before_install:
-  - 'sudo apt-get -qq update'
-  - 'pip install -r requirements.txt'
+  - sudo apt-get -qq update
```
Comment (DoomGerbil) [inline .travis.yml:5]: Did we want to apply our platform norm of exploding out short flags in scripts here?  EG

```suggestion
  - sudo apt-get --quiet --quiet update
```

File: defaults/main.yml [hunk 4]
```diff
@@ -34,7 +35,7 @@ buildkite_agent_expose_secrets: false
 buildkite_agent_builds_dir:
   Darwin: "{{ buildkite_agent_home_dir[ansible_os_family] }}/builds"
   Debian: "{{ buildkite_agent_home_dir[ansible_os_family] }}/builds"
-  Windows: "c:/b"  # long filenames will overflow
+  Windows: c:/b  # long filenames will overflow
```
Comment (DoomGerbil) [inline defaults/main.yml:38]: quote usage is inconsistent between the `Windows` entry and the others in this dict.

File: defaults/main.yml [hunk 5]
```diff
@@ -80,16 +81,16 @@ buildkite_agent_tags_from_gcp_labels: "false"
...
+buildkite_agent_build_number: "3651"
 buildkite_agent_systemd_override_template: buildkite-agent.override.conf.j2
 
 # Windows options
-buildkite_agent_nssm_exe: "C:/ProgramData/chocolatey/bin/nssm.exe"
-buildkite_agent_nssm_version: "2.24.101.20180116"
-buildkite_agent_platform: "amd64"
-buildkite_agent_windows_grant_admin: False
+buildkite_agent_nssm_exe: C:/ProgramData/chocolatey/bin/nssm.exe
+buildkite_agent_nssm_version: 2.24.101.20180116
```
Comment (DoomGerbil) [inline defaults/main.yml:91]: This is an nssm version from 2018 - is this still right?

File: handlers/main.yml [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
+    when: ansible_os_family == "Debian" and buildkite_agent_allow_service_startup[ansible_os_family]
 
-- name: restart-windows-buildkite
-  win_service:
-    name: "buildkite-agent"
-    state: restarted
-  when: ansible_os_family == "Windows" and buildkite_agent_allow_service_startup[ansible_os_family]
+  - name: restart-windows-buildkite
+    win_service:
+      name: buildkite-agent
```
Comment (DoomGerbil) [inline handlers/main.yml:10]: Huh - did either of these ever work with the bad indentation?

File: meta/main.yml [hunk 7]
```diff
@@ -1,20 +1,20 @@
...
+    - name: Ubuntu
       versions:
-        - 'focal'
-        - 'xenial'
-    - name: 'Windows'
+        - focal
+        - xenial
+    - name: Windows
       versions:
         - '2016'
```
Comment (DoomGerbil) [inline meta/main.yml:14]: Still `2016` only for Windows?

**Review by DoomGerbil — DISMISSED** [review]
Couple of comments/questions, LGTM

File: tasks/install-on-Debian.yml [hunk 8]
```diff
@@ -1,76 +1,77 @@
...
-    system: yes
-    uid: "{{ buildkite_agent_user_uid | default(omit) }}"
+  - name: Create the user to run the agent
+    user:
+      name: "{{ buildkite_agent_username }}"
+      create_home: true
+      group: "{{ buildkite_agent_username }}"
+      # because the deb creates /var/lib/buildkite-agent by default.
+      home: /var/lib/{{ buildkite_agent_username }}
+      shell: ""
```
Comment (PaulSonOfLars) [inline tasks/install-on-Debian.yml:14]: does this specifically set there to be no shell?

**Review by PaulSonOfLars — APPROVED** [review]

**Comment by DoomGerbil** [comment]
Going to close this out to tie up some loose ends - there've been enough changes in the interim that it probably needs re-doing from scratch at this stage.

**Pull request closed without merging**